From 11496890d7e21786cbfd9fd17ab968f498116b3f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Tue, 19 Jan 2016 10:17:20 +0100
Subject: [PATCH] Tell jQuery we're expecting JSON here
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

It's better to be explicit rather than relying on autodetection.

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 js/codemirror/addon/lint/sql-lint.js | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/js/codemirror/addon/lint/sql-lint.js b/js/codemirror/addon/lint/sql-lint.js
index 22a5785a6db..0a1f396f4b0 100644
--- a/js/codemirror/addon/lint/sql-lint.js
+++ b/js/codemirror/addon/lint/sql-lint.js
@@ -6,9 +6,7 @@ CodeMirror.sqlLint = function(text, updateLinting, options, cm) {
         return;
     }
 
-    function handleResponse(json) {
-        response = JSON.parse(json);
-
+    function handleResponse(response) {
         var found = [];
         for (var idx in response) {
             found.push({
@@ -29,6 +27,7 @@ CodeMirror.sqlLint = function(text, updateLinting, options, cm) {
     $.ajax({
         method: "POST",
         url: "lint.php",
+        dataType: 'json',
         data: {
             sql_query: text,
             token: PMA_commonParams.get('token'),
